src/tooltip.tsx

```diff
@@ -135,10 +139,77 @@ const Tooltip: React.FC<Props> = ({
         height: 0,
     });
 
-    const getPosition = (position: Position = defaultPositionDesktop) =>
-        isTabletOrSmaller && (position === 'left' || position === 'right') ? defaultPositionMobile : position;
+    const [containerPosition, setContainerPosition] = React.useState({});
 
-    const position = getPosition(rest.position);
+    const getPosition = React.useCallback(
+        (position: Position | undefined) =>
+            isTabletOrSmaller ? position || defaultPositionMobile : position || defaultPositionDesktop,
+        [isTabletOrSmaller]
+    );
+
+    const tooltipBoundingClientRect = React.useCallback(() => {
+        if (!tooltipRef.current) return undefined;
+
+        return tooltipRef.current.getBoundingClientRect();
+    }, [tooltipRef]);
+
+    const tooltipClient = tooltipBoundingClientRect();
+
+    const hasSpace = React.useCallback(
+        (position: Position) => {
+            if (!tooltipClient) return undefined;
+
+            return position === 'left'
+                ? targetBoundingClientRect.current.left < tooltipClient.width
+                : targetBoundingClientRect.current.right + tooltipClient.width > window.innerWidth;
+        },
+        [tooltipClient]
+    );
```

Comment:
what's the intention of this function? it says if there is enough space in the given position? then the logic doesn't seem to be taking into account the 'top' / 'bottom' cases. It seems to be handling only the right/left cases. If that's true, then type it as `(position: 'left' | 'right') =>`. Or even better, just create two different functions (`hasLeftSpace` and `hasRightSpace`). Also I think you aren't taking into account the space between target and tooltip in the calcs

Reply:
Ok, I separated it into two different functions, regarding the calculations, the part that takes into account the space between the target and the tooltip calcs is in another part of the code (I believe it is in containerPos), originally I had separated this verification to simplify how it is used in many different parts of the code in various calculations.